src/wasm-interpreter.h

```diff
@@ -2531,7 +2574,15 @@ template<typename GlobalManager, typename SubType> class ModuleInstanceBase {
       }
       NOTE_EVAL1(name);
       NOTE_EVAL1(flow.getSingleValue());
-      instance.globals[name] = flow.values;
+
+      auto* global = instance.wasm.getGlobal(name);
+      if (global->imported()) {
+        auto inst = instance.linkedInstances.at(global->module);
+        Export* globalExport = inst->wasm.getExport(global->base);
+        inst->globals[globalExport->value] = flow.values;
+      } else {
+        instance.globals[name] = flow.values;
+      }
```

Comment:
almost the same logic appears in GlobalGet and Set. How about a helper method that returns a reference to the value of the global?

Reply:
Good idea, Thanks! Done. I just don't have a good idea in case of imported tables, for the table instructions branch.